lib/galaxy/jobs/__init__.py

```diff
@@ -1743,7 +1743,14 @@ def _change_ownership( self, username, gid ):
         job = self.get_job()
         # FIXME: hardcoded path
         external_chown_script = self.get_destination_configuration("external_chown_script", None)
-        cmd = [ '/usr/bin/sudo', '-E', external_chown_script, self.working_directory, username, str( gid ) ]
+
+        cmd = [ '/usr/bin/sudo', '-E', ]
+        for v in ["PATH", "LD_LIBRARY_PATH", "PKG_CONFIG_PATH"]:
```

Comment:
This shouldn't be necessary, as 'sudo -E' does preserve the environment.
This is a small snippet that you can use to verify this:
```python
import os
import subprocess


print('without -E:')
subprocess.call(['sudo', './echo.sh'], shell=False)
print('with -E:')
subprocess.call(['sudo', '-E', './echo.sh'], shell=False)
```

Write a echo.sh script and echo some environment variable only available to the user.
You'll see that only with '-E' you get the environment variable echoed correctly.